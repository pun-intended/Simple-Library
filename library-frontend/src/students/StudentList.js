// Show all students in class -> Student Card
// Dropdown menu for class
import React from "react"
import StudentCard from "./StudentCard";
import { useState, useEffect, useContext } from "react";
import LibraryApi from "../api";
import {Container, Col, Row } from 'reactstrap'
import "./StudentCard.css";
import StudentContext from "../StudentContext";

const StudentList = () => {
    const [update, setUpdate] = useState(false)
    const {students, setStudents} = useContext(StudentContext)
    
    useEffect( () => {
        async function initializeList(){
            const allStudents = await LibraryApi.getAllStudents()
            setStudents(allStudents)
            setUpdate(false)
        }
        initializeList()
    }, [update])

    return(
        <div className="StudentList">
            <Container>
                <h1> All Students </h1>
                <Row>
            {students.map((st) =>  {
                return (
                    <Col xs="6" sm="4" md="3" lg="2">
                    <StudentCard student={st} setUpdate={setUpdate} key={st.id}/>
                    </Col>
                )})}
                </Row>
            </Container>
            
        </div>
    )
}

export default StudentList;