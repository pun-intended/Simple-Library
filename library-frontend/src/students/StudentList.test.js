import React from "react";
import { render, screen } from "@testing-library/react";
import StudentList from "./StudentList";
import StudentContext from "../StudentContext";


const students = {
    students: [{
        first_name: "test",
        last_name: "student",
        id: "1001"
    },
    {
        first_name: "test",
        last_name: "student",
        id: "1002",
        book_id: "101"
    }]
};

const initializeList = jest.fn()

// Smoke test
it("renders without crashing", () => {
    render(<StudentContext.Provider value={students}>
        <StudentList />
        </StudentContext.Provider>);
});

// Snapshot
it("Matches snapshot", () => {
    const {asFragment} = render(<StudentContext.Provider value={students}>
        <StudentList />
        </StudentContext.Provider>);
    expect(asFragment()).toMatchSnapshot();
})

// List contains all students
it("contains all students from test data", () => {
    render(
        <StudentContext.Provider value={students}>
        <StudentList />
        </StudentContext.Provider>);
    // expect(getAllByRole('heading').length).toEqual(students.students.length);
    expect(screen.getAllByRole('heading', {name: /test student/i}).length).toEqual(students.students.length)
})
