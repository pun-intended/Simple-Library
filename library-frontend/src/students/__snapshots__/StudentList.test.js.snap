// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="StudentList"
  >
    <div
      class="container"
    >
      <h1>
         All Students 
      </h1>
      <div
        class="row"
      >
        <div
          class="col-6 col-sm-4 col-md-3 col-lg-2"
        >
          <div
            class="StudentCard card"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h6
                  class="card-title"
                >
                  test student
                </h6>
              </div>
              <div
                class="col"
              >
<<<<<<< Updated upstream
                <svg
                  aria-labelledby="svg-inline--fa-title-z7tTqHVQndN6"
                  class="svg-inline--fa fa-hand-holding btn btn-primary"
                  data-icon="hand-holding"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="svg-inline--fa-title-z7tTqHVQndN6"
                  >
                    no-book
                  </title>
                  <path
                    d="M559.7 392.2c17.8-13.1 21.6-38.1 8.5-55.9s-38.1-21.6-55.9-8.5L392.6 416H272c-8.8 0-16-7.2-16-16s7.2-16 16-16h16 64c17.7 0 32-14.3 32-32s-14.3-32-32-32H288 272 193.7c-29.1 0-57.3 9.9-80 28L68.8 384H32c-17.7 0-32 14.3-32 32v64c0 17.7 14.3 32 32 32H192 352.5c29 0 57.3-9.3 80.7-26.5l126.6-93.3zm-366.1-8.3a.5 .5 0 1 1 -.9 .2 .5 .5 0 1 1 .9-.2z"
                    fill="currentColor"
                  />
                </svg>
=======
                <button
                  class="btn btn-primary btn btn-secondary"
                  type="button"
                >
                   Check out 
                </button>
>>>>>>> Stashed changes
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-6 col-sm-4 col-md-3 col-lg-2"
        >
          <div
            class="StudentCard card"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h6
                  class="card-title"
                >
                  test student
                </h6>
              </div>
              <div
                class="col"
              >
<<<<<<< Updated upstream
                <svg
                  aria-labelledby="svg-inline--fa-title-Rr1R9kZcD4ev"
                  class="svg-inline--fa fa-book-open fa-lg btn btn-primary"
                  data-icon="book-open"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="svg-inline--fa-title-Rr1R9kZcD4ev"
                  >
                    has-book
                  </title>
                  <path
                    d="M249.6 471.5c10.8 3.8 22.4-4.1 22.4-15.5V78.6c0-4.2-1.6-8.4-5-11C247.4 52 202.4 32 144 32C93.5 32 46.3 45.3 18.1 56.1C6.8 60.5 0 71.7 0 83.8V454.1c0 11.9 12.8 20.2 24.1 16.5C55.6 460.1 105.5 448 144 448c33.9 0 79 14 105.6 23.5zm76.8 0C353 462 398.1 448 432 448c38.5 0 88.4 12.1 119.9 22.6c11.3 3.8 24.1-4.6 24.1-16.5V83.8c0-12.1-6.8-23.3-18.1-27.6C529.7 45.3 482.5 32 432 32c-58.4 0-103.4 20-123 35.6c-3.3 2.6-5 6.8-5 11V456c0 11.4 11.7 19.3 22.4 15.5z"
                    fill="currentColor"
                  />
                </svg>
=======
                <button
                  class="btn btn-primary btn btn-secondary"
                  type="button"
                >
                   Check in 
                </button>
>>>>>>> Stashed changes
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
